### ISSUE TITLE
<collection>.where and .first_where consistency

### ISSUE THREAD

**Issue by gskip17** [periscope-ps/UNISrt#33]
Given you get a valid arbitrary node out of unis.
```
n = rt.nodes[0]
```
You cannot `.where` or `.first_where` its internal collection, although the documentation says nothing about these functions being strictly for top level resources. In theory these functions should work on any initialized Collection object.

Example -
```
# Pretty much every variation of this will work (first_where, lambda, dict, etc)
n = rt.nodes.where(lambda node: node.name == "my_node") 
```
------------
```
# Every variation of the where function set fails looking at an internal collection
p = n.ports.where(lambda port: port.name == "what im looking for")
```

Traceback -
```
>>> n.ports.where({"properties":{"vport_number:":120}})  
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/gskipper/UNISrt/unis/models/models.py", line 52, in f  
    return v(*args, **kwargs)
  File "/home/gskipper/UNISrt/unis/models/models.py", line 306, in where   
      if isinstance(pred, types.FunctionType):
NameError: name 'pred' is not defined
```

Now, I get there are a ton of ways to skin this cat. So it not a big issue. Just wanted to point it out.

**Comment by jemusser** [comment]
This functionality hasn't had much stress testing, it's probably an
old-fashioned bug.  I'll take a look at it as soon as I have a free cycle.

**Comment by jemusser** [comment]
Resolved in 29d1e1c27fb740a22ae13883c1fd181e7b730eef
